WalletWasabi/WabiSabi/Client/ArenaClient.cs

```diff
@@ -114,6 +114,31 @@ public async Task RegisterOutputAsync(Guid roundId, long value, Script scriptPub
 			WeightCredentialClient.HandleResponse(outputRegistrationResponse.WeightCredentials, realWeightCredentialResponseValidation);
 		}
 
+		public async Task ReissuanceAsync(Guid roundId, long value, IEnumerable<Credential> amountCredentialsToPresent)
+		{
+			Guard.InRange(nameof(amountCredentialsToPresent), amountCredentialsToPresent, 0, AmountCredentialClient.NumberOfCredentials);
+
+			var presentedAmount = amountCredentialsToPresent.Sum(x => (long)x.Amount.ToUlong());
+
+			var amount1 = value;
+			var amount2 = presentedAmount - value;
```

Comment:
This variables are not strictly necessary and I would remove them.

Reply:
As always, this is just a suggestion.

Reply:
If I remove this, I will get an exception that the 
https://github.com/molnard/WalletWasabi/blob/2206e5cc916724b8fe21a97af891c72dfdb816b1/WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs#L471

Reply:
Added back, the two amounts. If I won't specify the second amount then I won't have two valuable creds, just one.

Reply:
Oh sorry, I didn't expressed myself well enough, what i meant is that the variables were not necessary but the values are.

Reply:
I think it makes more sense to specify all `k`/2 values, and ensure they sum to exactly the presented amount, rather than specifying `k-1` and computing the last one, i.e. instead of `long value` `IEenumerable<long> values` or something like that.

Reply:
ACK, added a check.

Reply:
I value2 can always be equal to the sum - value1 then does it makes sense to have it as a parameter?

What the credissuer does if the presented are more then I required. Are those creds lost? Like I present 5 and reissue 2,2 then 1 will be lost?

Reply:
The same applies to vSize creds. If I am requesting 2 x `scriptPubKey.EstimateOutputVsize()` then the Delta won't be zero.

Reply:
the delta must always be 0, otherwise the request is rejected, so technically the last value contains no information, but i think it's going to be simpler to just require the full list of values to be specified so there is a 1:1 correspondence between what the higher level round client specifies and what it receives in return.